test/integration/controller/jobs/raycluster/raycluster_controller_test.go

```diff
@@ -0,0 +1,730 @@
+	})
+
+	ginkgo.It("Should reconcile RayClusters", func() {
+		ginkgo.By("checking the job gets suspended when created unsuspended")
+		priorityClass := testing.MakePriorityClass(priorityClassName).
+			PriorityValue(priorityValue).Obj()
+		gomega.Expect(k8sClient.Create(ctx, priorityClass)).Should(gomega.Succeed())
+
+		job := testingraycluster.MakeCluster(jobName, ns.Name).
+			Suspend(false).
+			WithPriorityClassName(priorityClassName).
+			Obj()
+		err := k8sClient.Create(ctx, job)
+		gomega.Expect(err).To(gomega.Succeed())
+		createdJob := &rayv1.RayCluster{}
+
+		setInitStatus(jobName, ns.Name)
+
+		gomega.Eventually(func() bool {
+			if err := k8sClient.Get(ctx, types.NamespacedName{Name: jobName, Namespace: ns.Name}, createdJob); err != nil {
+				return false
+			}
+			return *createdJob.Spec.Suspend
+		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+
+		ginkgo.By("checking the workload is created without queue assigned")
+		createdWorkload := &kueue.Workload{}
+		gomega.Eventually(func() error {
+			return k8sClient.Get(ctx, wlLookupKey, createdWorkload)
+		}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		gomega.Expect(createdWorkload.Spec.QueueName).Should(gomega.Equal(""), "The Workload shouldn't have .spec.queueName set")
+		gomega.Expect(metav1.IsControlledBy(createdWorkload, createdJob)).To(gomega.BeTrue(), "The Workload should be owned by the Job")
+
+		ginkgo.By("checking the workload is created with priority and priorityName")
+		gomega.Expect(createdWorkload.Spec.PriorityClassName).Should(gomega.Equal(priorityClassName))
+		gomega.Expect(*createdWorkload.Spec.Priority).Should(gomega.Equal(priorityValue))
+
+		ginkgo.By("checking the workload is updated with queue name when the job does")
+		jobQueueName := "test-queue"
+		createdJob.Annotations = map[string]string{constants.QueueAnnotation: jobQueueName}
+		gomega.Expect(k8sClient.Update(ctx, createdJob)).Should(gomega.Succeed())
+		gomega.Eventually(func() bool {
+			if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+				return false
+			}
+			return createdWorkload.Spec.QueueName == jobQueueName
+		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+
+		// TODO
+		// changes are tested with master version of ray using go get github.com/ray-project/kuberay/ray-operator@master
+		// Need to revisit this after the new released version of ray since I had to add scheme for v1 inside
+		// kueue/client-go/clientset/versioned/scheme/register.go
+		ginkgo.By("checking a second non-matching workload is deleted")
+		secondWl := &kueue.Workload{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      workloadraycluster.GetWorkloadNameForRayCluster("second-workload"),
+				Namespace: createdWorkload.Namespace,
+			},
+			Spec: *createdWorkload.Spec.DeepCopy(),
+		}
+
+		gomega.Expect(ctrl.SetControllerReference(createdJob, secondWl, scheme.Scheme)).Should(gomega.Succeed())
```

Comment:
This is not the same scheme as the test scheme where types are registered into.

The `sigs.k8s.io/kueue/client-go/clientset/versioned/scheme` import should be replaced with `k8s.io/client-go/kubernetes/scheme` instead.